### PR TITLE
[PRMDR-466]

### PR DESCRIPTION
**Pull request by AlexHerbertNHS**
Adding NHS number validation to the one place it was missing and adding a test
Fixing some sonar stuff on LG validator

### CODE REVIEW

File: lambdas/tests/unit/handlers/test_create_document_reference_handler.py [hunk 1]
```diff
@@ -92,11 +92,11 @@ def test_lambda_handler_missing_environment_variables_type_lg_returns_400(
...
-    set_env,
-    monkeypatch,
-    arf_type_event,
-    environment_variable,
-    context,
+        set_env,
+        monkeypatch,
+        arf_type_event,
+        environment_variable,
+        context,
```
Comment (abbas-khan10) [inline lambdas/tests/unit/handlers/test_create_document_reference_handler.py:99]: Did you run `make format` before pushing? These changes to the format look a bit strange

File: lambdas/handlers/create_document_reference_handler.py [hunk 2]
```diff
@@ -58,6 +59,10 @@ def processing_event_details(event):
     try:
         body = json.loads(event["body"])
         nhs_number = body["subject"]["identifier"]["value"]
+        try:
+            validate_id(nhs_number)
+        except InvalidResourceIdException:
+            raise CreateDocumentRefException(400, "Invalid NHS number")
```
Comment (NogaNHS) [inline lambdas/handlers/create_document_reference_handler.py:65]: `validate_id` is already used in the create reference service, as it is part of the business logic.

File: lambdas/utils/lloyd_george_validator.py [hunk 3]
```diff
@@ -167,7 +168,7 @@ def validate_with_pds_service(file_name_list: list[str], nhs_number: str):
         if patient_details.general_practice_ods != current_user_ods:
             raise LGInvalidFilesException("User is not allowed to access patient")
 
-    except (ValidationError, ClientError, ValueError) as e:
+    except (ClientError, ValueError) as e:
```
Comment (NogaNHS) [inline lambdas/utils/lloyd_george_validator.py:171]: What is the reason for removing `ValidationError`?

Comment (AlexHerbertNHS) [reply lambdas/utils/lloyd_george_validator.py:171]: ValidationError inherits from ValueError, so it'll be caught by this either way

File: lambdas/tests/unit/handlers/test_create_document_reference_handler.py (hunk 1 above)
Comment (AlexHerbertNHS) [reply lambdas/tests/unit/handlers/test_create_document_reference_handler.py]: I guess this is my IDE formatting it

When I do make format, I get 49 files changed:
image

**Pull request closed without merging**